sync-core/src/main/java/com/cloudant/sync/replication/BasicPullStrategy.java

```diff
@@ -387,6 +387,10 @@ private ChangesResultWrapper nextBatch() throws DatastoreException {
 
         List<Callable<DocumentRevsList>> tasks = new ArrayList<Callable<DocumentRevsList>>();
         for(String id : ids) {
+            //skip any document with an empty id
+            if(id.isEmpty()){
+                continue;
+            }
```

Comment:
Possibly output a debug log message here to say that we saw a changes feed entry with an empty ID and are ignoring it.